fix: support validation of stringToString values in ConfigMap

alex-berger opened the pull request:
This is the second attempt to fix https://github.com/cilium/cilium/issues/33095, the first attempt lead to regression https://github.com/cilium/cilium/issues/34129 and was reverted in https://github.com/cilium/cilium/pull/34277.

 This time using command.ToStringMapStringE(...) instead of cast.ToStringMapStringE(...) to properly support non JSON encoded, comma separated key value encoding.

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #33095

```release-note
add support for validation of stringToString values in ConfigMap
```

derailed reviewed (commented):
@alex-berger Thank you for this update! As Joe mentioned it would be good to have some tests on this to avoid regressions. Perhaps validateConfigMap should be modified to take in FlagSet vs the entire cobra cmd might be a step in the right direction??

joestringer reviewed (approved):
I like this even better with @derailed's suggestion since the validation doesn't need to cross over to the daemon package.

Awesome, thanks for putting this together :pray:

ldelossa reviewed (approved):
LGTM, thanks

alex-berger committed:
fix: support validation of stringToString values in ConfigMap

Fixes:#33095

This is the second attempt to fix https://github.com/cilium/cilium/issues/33095,
the first attempt lead to regression https://github.com/cilium/cilium/issues/34129
and was reverted in https://github.com/cilium/cilium/pull/34277.

Using command.ToStringMapStringE(...) instead of cast.ToStringMapStringE(...)

Signed-off-by: Alexander Berger <alex-berger@gmx.ch>

alex-berger committed:
fix: add regression tests for api-rate-limit validation

Signed-off-by: Alexander Berger <alex-berger@gmx.ch>
